spec/models/acts_as_customizable/calculated_value_spec.rb

```diff
@@ -381,4 +382,194 @@ def custom_field_values(*) = nil
+      let(:cv3) { create(:calculated_value_project_custom_field, :skip_validations, projects: [project], formula: "5 + 5") }
+
+      let(:enabled_custom_fields) do
+        { cv1 => nil, cv2 => nil, cf_int => nil, cv3 => nil }
+      end
+
+      it "creates a missing value error" do
+        project.calculate_custom_fields([cv1, cv2, cv3])
+
+        expect(cv3.calculated_value_errors.where(customized: project)).to be_empty
+
+        expect_calculated_value_error(cv1, project, "ERROR_MISSING_VALUE", cf_int.name)
+        expect_calculated_value_error(cv2, project, "ERROR_MISSING_VALUE", cv1.name)
+      end
+    end
+
+    describe "disabled value" do
+      let(:cf_int) { create(:integer_project_custom_field) }
+      let(:cv1) do
+        create(:calculated_value_project_custom_field, :skip_validations, projects: [project], formula: "5 + #{cf_int}")
+      end
+      let(:cv2) { create(:calculated_value_project_custom_field, :skip_validations, projects: [project], formula: "5 + #{cv1}") }
+      let(:cv3) { create(:calculated_value_project_custom_field, :skip_validations, projects: [project], formula: "5 + 5") }
+
+      let(:enabled_custom_fields) do
+        { cv1 => nil, cv2 => nil, cv3 => nil }
+      end
+
+      it "creates a disabled value error" do
+        project.calculate_custom_fields([cv1, cv2, cv3])
+
+        expect(cv3.first_calculation_error(project)).to be_blank
+
+        # The referenced int field is disabled, we thus expect a `disabled` error.
+        expect_calculated_value_error(cv1, project, "ERROR_DISABLED_VALUE", cf_int.name)
+
+        # Since cv1 cannot be calculated, it leads to a missing value error in cv2.
+        expect_calculated_value_error(cv2, project, "ERROR_MISSING_VALUE", cv1.name)
+      end
+    end
+
+    context "with disabled calculated values" do
+      let(:cf_int) { create(:integer_project_custom_field) }
+      let(:formula) { "2 / #{cf_int}" }
+      let(:disabled_cv) do
+        create(:calculated_value_project_custom_field, :skip_validations, projects: [], formula:)
+      end
+      let(:enabled_cv) { create(:calculated_value_project_custom_field, :skip_validations, projects: [project], formula:) }
+
+      let(:enabled_custom_fields) do
+        { enabled_cv => nil, cf_int => 0 }
+      end
+
+      it "only creates errors for enabled calculated values" do
+        project.calculate_custom_fields([disabled_cv, enabled_cv])
+
+        expect_calculated_value_error(enabled_cv, project, "ERROR_MATHEMATICAL")
+        expect(disabled_cv.first_calculation_error(project)).to be_blank
+      end
+    end
+
+    describe "nesting multiple custom fields into one formula" do
+      let(:a) { create(:integer_project_custom_field) }
+      let(:b) { create(:integer_project_custom_field) }
+      let(:a_plus_b) do
+        create(:calculated_value_project_custom_field, :skip_validations, projects: [project], formula: "#{a} + #{b}")
```

Comment:
Probably easier to follow with additional `let(:a_plus_b_enabled)` 
```suggestion
        create(:calculated_value_project_custom_field, :skip_validations, projects: a_plus_b_enabled ? [project] : [], formula: "#{a} + #{b}")
```